src/components/form/formFieldFactory.js

```diff
@@ -135,6 +135,14 @@ function customFieldComponent(customField, cfg) {
             ...baseConfig,
             values: customField.selectableValues
         })
+    } else if (customField.dataType === 'email') {
+        return simpleField({
+            ...baseConfig,
+            type: 'email',
+            validator: email
+        });
+    } else if (customField.dataType === 'phone') {
+        return simpleField(baseConfig)
     } else if (customField.dataType === 'string') {
         return simpleField(baseConfig)
```

Comment:
```suggestion
    } else if (customField.dataType === 'phone' || customField.dataType === 'string') {
        return simpleField(baseConfig)
```

Reply:
I'd rather have two `if else`conditions for the `phone` and `string` types. It's easier to understand and `string` is like the `default` type.

Reply:
Although I find your proposal valuable, I actually prefer keeping the former "pattern-match" looking format, at least for 2 reasons:
- IMO vertical stacking is more readable and extendable (what if a third then a forth string-based type is added)
- phone type should have it's own validator (following what was done for the email)